acknowledge admin and course team on program page

tasawernawaz opened the pull request:
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [ ] Desktop screenshots
  - [ ] Mobile width screenshots
  - [ ] Tag @ferdi or @pdpinch for review
- [x] Migrations
  - [ ] Migration is backwards-compatible with current production code
- [x] Testing
  - [ ] Code is tested
  - [ ] Changes have been manually tested
- [ ] Settings
  - [ ] New settings are documented and present in `app.json`
  - [ ] New settings have reasonable development defaults, if applicable

#### What are the relevant tickets?
Fixes #4430 

#### What's this PR do?
Add course team tab page for program page.

#### How should this be manually tested?

- Goto cms
- Add child page "Course team"
- Fill the different sections on the page.
- Goto to program page
- You should see a new tab page for the course team.

#### Screenshots (if appropriate)
**CMS**
![68310402-c4e92500-00d1-11ea-9e05-d8ff7e9e105e](https://user-images.githubusercontent.com/4245618/68486708-7cfa0780-0263-11ea-8ba2-dbf6aaffb234.png)


**Rendering**
![screencapture-localhost-8079-my-program-page-course-team-2019-11-08-19_40_30](https://user-images.githubusercontent.com/4245618/68486616-4623f180-0263-11ea-8b26-70b89acbacde.png)

tasawernawaz commented:
@pdpinch Could you please take a quick look at the design, you might want to change the font color of the instructor's name on the image.

tasawernawaz commented:
FYI @abdulkdawson ^

abdulkdawson commented:
@tasawernawaz what are the image dimensions?

abdulkdawson commented:
@tasawernawaz Also can we add a black gradient to the bottom of the image so that the name is clearer. I'd rather do that then change the font color. It can be just as problematic if we do that.

pdpinch commented:
The Instructor images should be larger than the administrator images. @tasawernawaz Have you been able to access [inspect mode](https://projects.invisionapp.com/d/main/#/console/18769740/391397475/inspect)?

tasawernawaz commented:
@pdpinch @abdulkdawson Sorry I missed the difference in profile image of instructors and administrators. I have updated the design with the given dimensions of the image and added gradient too. However, the instructor section given dimension is 385x385 which can only be seen when you have a high-resolution screen. New screenshot attached below.

![screencapture-localhost-8079-my-program-page-course-team-2019-11-11-18_20_42](https://user-images.githubusercontent.com/4245618/68590439-495fed00-04b0-11ea-97e5-acb005608d76.png)

tasawernawaz commented:
@pdpinch I have following question about adding `a link to this "course team" page below the existing faculty carousel (conditional on whether the page exists)`

Q#1: Will link to the course team be on every tab as we have faculty carousel on every tab.
Q#2: What should the be design to add a link to the course team.

ahmed-arbisoft commented on cms/blocks.py:
```diff
@@ -0,0 +1,16 @@
+    """
+
+    name = blocks.CharBlock(max_length=100, help_text="Name of the course team member.")
+    bio = blocks.CharBlock(max_length=255, help_text="Short bio of course team member.")
```
Are we okay with the 255 character limit on this?

tasawernawaz replied:
@pdpinch could you please answer this question.

pdpinch replied:
Yes, definitely. The size of the input box in the CMS is too small.

pdpinch replied:
Also, 255 characters is not enough for the existing sample copy at https://drive.google.com/file/d/1hYhbeNmzSVS-Hd3zBNzhcsRd2uefIlcJ/view 

The longest bio there looks to me to be about 700 characters. I'm not sure if this breaks the layout.

tasawernawaz commented:
@pdpinch I know that @annagav is on leave, who else can review this? @ahmed-arbisoft has reviewed but he cannot do a functional test as he doesn't have MM locally.

pdpinch commented:
We decided not to do [the link](https://github.com/mitodl/micromasters/pull/4454#issuecomment-552831602) for now. If we still need to do it, I'll open another issue. 

I updated the issue description.

tasawernawaz commented:
> We decided not to do [the link](https://github.com/mitodl/micromasters/pull/4454#issuecomment-552831602) for now. If we still need to do it, I'll open another issue.
> 
> I updated the issue description.

thanks, what else do think required on this before we can merge this one.

pdpinch commented:
Can you set up the review app so I can interact with the page? I'd like to see how the bios work on hover.

tasawernawaz commented:
> Can you set up the review app so I can interact with the page? I'd like to see how the bios work on hover.

sure thing.

tasawernawaz commented:
@pdpinch I have setup a review app for testing.
You can review at https://micromasters-ci-pr-4454.herokuapp.com/test-course-team-program/
You can edit at https://micromasters-ci-pr-4454.herokuapp.com/cms/pages/5/edit/

User: `staff`
password: `edx`

annagav commented:
@tasawernawaz, I noticed that when you hover over the picture the name of faculty changes font. Is that intentional? 
Also, something feels abrupt about the hover. Is it possible to make the fade out of the picture more smooth. If this feature is urgent, then it's okay as it is.

pdpinch commented:
@annagav I get an animation of the bio when I hover -- it scrolls up. What browser are you using?

annagav commented:
@pdpinch yes, I can see the hover, but I thought something looked out of sync , or happens too fast. Like the fade out of the image good have a smoother transition, the size of the font changes abruptly.

pdpinch commented:
Please make the layout responsive, and include a screenshot at mobile width:

This isn't a good look: 

![image](https://user-images.githubusercontent.com/430126/68797862-4e917900-0623-11ea-8628-658f07ea2434.png)

pdpinch commented:
Also, I'm not sure why we have so much padding between the course administrator images. It would be nice to allow them to go to four across at desktop screen size.

pdpinch commented:
The course team tab is displayed on the "about the progra" page, but it disappears on the "course team" page. It should change to a selected state, similar to what the FAQ does on the current site. 

![image](https://user-images.githubusercontent.com/430126/68798220-ff981380-0623-11ea-876c-87facfdf5c03.png)

pdpinch commented:
@tasawernawaz for Anna's [animation question](https://github.com/mitodl/micromasters/pull/4454#issuecomment-553559032), can we slow it down?

tasawernawaz commented:
@pdpinch @annagav I can add animation, but difference in the font and padding between administrators are intentional as they are given in the design https://invis.io/Y8UKRZZSC6D#/391397475_MicroMasters_Team_Page_Copy.

Honestly, It wasn't an easy design, I have to took help from a designer here to match the exact padding on different sections of the page.

tasawernawaz commented:
@annagav , @pdpinch I have addressed the following comments

- added opacity to handle hover state better.
- updated administrators section to have four items
- Fix active tab issue
- Fix for mobile and tablet

**Desktop**
![screencapture-mm-odl-local-8079-test-tasawer-program-course-team-2019-11-14-15_45_58](https://user-images.githubusercontent.com/4245618/68851171-34c16600-06f7-11ea-9b7f-1c88ec0172ff.png)

**Tablet**
![screencapture-mm-odl-local-8079-test-tasawer-program-course-team-2019-11-14-15_49_02](https://user-images.githubusercontent.com/4245618/68851194-40149180-06f7-11ea-87a6-343ff1754e89.png)


**Mobile**
![screencapture-mm-odl-local-8079-test-tasawer-program-course-team-2019-11-14-15_59_10](https://user-images.githubusercontent.com/4245618/68851454-cf21a980-06f7-11ea-804e-db67cf4ad457.png)
